### PR TITLE
fix: add index on oc_filecache over size, parent and storage to speed…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
… up Cache::calculateFolderSize
## Description
Folder sizes are computed quite often - especially on massive file upload.

### Before
```
-> Aggregate: sum(oc_filecache.size), min(oc_filecache.size)  (cost=1.26 rows=1) (actual time=0.118..0.119 rows=1 loops=1)
    -> Filter: (oc_filecache.`storage` = 1)  (cost=1.25 rows=0.05) (actual time=0.109..0.112 rows=5 loops=1)
        -> Index lookup on oc_filecache using fs_parent_name_hash (parent=1), with index condition: (oc_filecache.parent = 1)  (cost=1.25 rows=5) (actual time=0.108..0.110 rows=5 loops=1)
```

### After
```
-> Aggregate: sum(oc_filecache.size)  (cost=1.63 rows=1) (actual time=0.013..0.013 rows=1 loops=1)
    -> Filter: (oc_filecache.parent = 1)  (cost=1.13 rows=5) (actual time=0.008..0.011 rows=5 loops=1)
        -> Covering index lookup on oc_filecache using fs_parent_storage_size (parent=1, storage=1)  (cost=1.13 rows=5) (actual time=0.006..0.008 rows=5 loops=1)

```

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/5595

## How Has This Been Tested?
- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: core/Migrations/Version20230210073645.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use OCP\Migration\ISchemaMigration;
+
+class Version20230210073645 implements ISchemaMigration {
+	/**
+	 * @throws SchemaException
+	 */
+	public function changeSchema(Schema $schema, array $options): void {
+		$prefix = $options['tablePrefix'];
+		$table = $schema->getTable("{$prefix}filecache");
+		$table->addIndex(['size', 'parent', 'storage']);
```
Comment (IljaN) [inline core/Migrations/Version20230210073645.php:15]: People from SWITCH and SURF propose to change the order and put size last to optimize the index further

**Review by IljaN — CHANGES_REQUESTED** [review]

File: core/Migrations/Version20230210073645.php (hunk 1 above)
Comment (IljaN) [reply core/Migrations/Version20230210073645.php:15]: Also keep in mind that the index needs to be updated if size changes.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIFilesExt-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37920/149

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-git-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37920/152

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt2-latest-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/37920/156

**Review by IljaN — APPROVED** [review]

**Pull request merged**